modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/GridDhtPartitionTopologyImpl.java

```diff
@@ -596,7 +596,8 @@ else if (localNode(p, aff))
 
                 long updateSeq = this.updateSeq.incrementAndGet();
 
-                cntrMap.clear();
+                if (!exchFut.firstEvent().eventNode().isClient() || exchFut.changedAffinity())
```

Comment:
Shouldn't the condition be

if (exchFut.exchangeType() == ALL && !exchFut.rebalanced())
      cntrMap.clear();

Reply:
Yes, you are absolutely correct.